pulsar-testclient/src/main/java/com/yahoo/pulsar/testclient/ModularLoadManagerBrokerMonitor.java

```diff
@@ -0,0 +1,206 @@
+            try {
+                newBrokers.addAll(zkClient.getChildren(path, this));
+            } catch (Exception ex) {
+                throw new RuntimeException(ex);
+            }
+            for (String oldBroker : brokers) {
+                if (!newBrokers.contains(oldBroker)) {
+                    System.out.println("Lost broker: " + oldBroker);
+                }
+            }
+            for (String newBroker : newBrokers) {
+                if (!brokers.contains(newBroker)) {
+                    System.out.println("Gained broker: " + newBroker);
+                    final BrokerDataWatcher brokerDataWatcher = new BrokerDataWatcher(zkClient);
+                    brokerDataWatcher.printBrokerData(path + "/" + newBroker);
+                }
+            }
+            this.brokers = newBrokers;
+        }
+    }
+
+    private static class BrokerDataWatcher implements Watcher {
+        private final ZooKeeper zkClient;
+
+        public BrokerDataWatcher(final ZooKeeper zkClient) {
+            this.zkClient = zkClient;
+        }
+
+        public static String brokerNameFromPath(final String path) {
+            return path.substring(path.lastIndexOf('/') + 1);
+        }
+
+        public synchronized void process(final WatchedEvent event) {
+            try {
+                if (event.getType() == Event.EventType.NodeDataChanged) {
+                    final String broker = brokerNameFromPath(event.getPath());
+                    printBrokerData(event.getPath());
+                }
+            } catch (Exception ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+
+        private static void printMessageData(final double msgThroughputIn, final double msgThroughputOut,
+                final double msgRateIn, final double msgRateOut) {
+            System.out.format("Message Throughput In: %.2f KB/s\n", msgThroughputIn / 1024);
+            System.out.format("Message Throughput Out: %.2f KB/s\n", msgThroughputOut / 1024);
+            System.out.format("Message Rate In: %.2f msgs/s\n", msgRateIn);
+            System.out.format("Message Rate Out: %.2f msgs/s\n", msgRateOut);
+        }
+
+        public synchronized void printBrokerData(final String brokerPath) {
+            final String broker = brokerNameFromPath(brokerPath);
+            final String timeAveragePath = ModularLoadManagerImpl.TIME_AVERAGE_BROKER_ZPATH + "/" + broker;
+            LocalBrokerData localBrokerData;
+            try {
+                localBrokerData = gson.fromJson(new String(zkClient.getData(brokerPath, this, null)),
+                        LocalBrokerData.class);
+            } catch (Exception ex) {
+                throw new RuntimeException(ex);
+            }
+
+            System.out.println("\nBroker Data for " + broker + ":");
+            System.out.println("---------------");
+
+            System.out.println("\nNum Topics: " + localBrokerData.getNumTopics());
```

Comment:
That's true, though the advantage of using a logger is that you can chose to change levels or save to a file with configuration. Also, every line will get the timestamp printed